src/engine.c

```diff
@@ -278,6 +278,11 @@ int init_context(struct app_config *app_config, struct supervisor_context *ctx)
   }
 
   utarray_new(ctx->config_ifinfo_array, &config_ifinfo_icd);
+  if (app_config->config_ifinfo_array == NULL) {
+    errno = EINVAL;
+    log_err("Invalid app config, missing config_ifinfo_array");
+    return -1;
```

Comment:
For most of your error handling you use `log_trace()` to print errors.

I felt like `log_err()` made more sense for an error condition (especially since `log_trace()` isn't enabled by default unless you change the logging levels), but you're welcome to change this to `log_trace()` instead if you feel like this is better.

Reply:
Thanks. I will have to change to log_trace, because log_err uses errno.

Reply:
Isn't `log_err` appropriate in this situation? I set the `errno` to [EINVAL, aka "Invalid argument"](https://en.cppreference.com/w/cpp/error/errno_macros#:~:text=EINVAL)

`log_trace()` is the lowest level of logging, below even `log_debug()`, so it only appears if you do `-dddddd`.

Reply:
Yes, one can use it like that. But a created `log_err` to be used only for system error that come out of glib library commands or kernel call. For engine.c `log_debug` would be the right one to use.